File: app/src/main/java/com/fafu/app/dfb/http/HeaderInterceptor.java
package com.fafu.app.dfb.http;

import androidx.annotation.NonNull;

import com.fafu.app.dfb.util.SPUtils;

import java.io.IOException;

import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

public class HeaderInterceptor implements Interceptor {

    @NonNull
    @Override
    public Response intercept(@NonNull Chain chain) throws IOException {
        Request.Builder builder = chain.request()
                .newBuilder()
                .addHeader("Prama", "no-cache")
                .addHeader("Cache-Control", "no-cache");
        if (SPUtils.get("Const").contain("User-Agent")) {
            builder.addHeader("User-Agent", SPUtils.get("Const").getString("User-Agent"));
        }
        return chain.proceed(builder.build());
    }
}

File: app/src/main/java/com/fafu/app/dfb/http/HeaderInterceptor.kt
package com.fafu.app.dfb.http

import com.fafu.app.dfb.util.SPUtils
import okhttp3.Interceptor
import okhttp3.Response

class HeaderInterceptor : Interceptor {
    override fun intercept(chain: Interceptor.Chain): Response {
        val requestBuilder = chain
                .request()
                .newBuilder()
                .addHeader("Pragma", "no-cache")
                .addHeader("Cache-Control", "no-cache")
        if (SPUtils.contain("User-Agent")) {
            requestBuilder.addHeader("User-Agent", SPUtils.getString("User-Agent"))
        }
        return chain.proceed(requestBuilder.build())
    }
}